### DISCUSSION TITLE
"PASSWORD_RECOVERY" Event not being fired (JS Client)

### DISCUSSION THREAD

**Discussion by abdulrahimiliasu** [supabase/supabase#14181] [Questions]
Hi,

Maybe I'm missing something here, but only the `"SIGNED_IN"` event only get's triggered after clicking on the reset password link.
Few questions I have:
1. Can we request password change several times within short period of time ? I noticed sometimes the `PASSWORD_RECOVERY` event get's fired and sometimes not.
2. Can I just use the `recovery_sent_at` user field to compare with the current time to see if the link has expired ? that way I don't have to check for any events. (if the user is signed in using the link of course)

here is my code,  Thanks.

```typescript
useEffect(() => {
        if (user) handleLoadUserData();

        supabaseClient.auth.onAuthStateChange(async (event) => {
            alert(event);
            if (event == "PASSWORD_RECOVERY") {
                setAppMetadata({ isPasswordRecoveryConfirmed: true });
            }
        });
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [user]);
```

**Comment by pratiknkulkarni** [comment]
I am facing the same issue, do we have any updates on this?

**Reply by abdulrahimiliasu** [reply]
They've fixed the issue, just update your package to the latest version

**Comment by nassmim** [comment]
still having the same issue. I am on
supabase/ssr  0.5.1
supabase/supabase-js 2.45.3

Any news?
